cmd/collector/app/span_processor.go

```diff
@@ -178,9 +178,17 @@ func (sp *spanProcessor) processItemFromQueue(item *queueItem) {
 }
 
 func (sp *spanProcessor) addCollectorTags(span *model.Span) {
-	// TODO add support for deduping tags, https://github.com/jaegertracing/jaeger/issues/1778
+	dedupKey := make(map[string]struct{})
+	for _, tag := range span.Process.Tags {
+		if value, ok := sp.collectorTags[tag.Key]; ok && value == tag.AsString() {
+			dedupKey[tag.Key] = struct{}{}
```

Comment:
I feel like a debug-level log entry would be suitable here.

Reply:
Done